File: pdu/python/hako_srv_msgs/pdu_cdr_conv_ServiceResponseHeader.py
from .pdu_pytype_ServiceResponseHeader import ServiceResponseHeader
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_ServiceResponseHeader(writer: CdrWriter, src: ServiceResponseHeader):
    writer.write_uint32(src.request_id)
    writer.write_string(src.service_name)
    writer.write_string(src.client_name)
    writer.write_uint8(src.status)
    writer.write_uint8(src.processing_percentage)
    writer.write_int32(src.result_code)


def cdr_body_to_py_ServiceResponseHeader(reader: CdrReader, dst: ServiceResponseHeader):
    dst.request_id = reader.read_uint32()
    dst.service_name = reader.read_string()
    dst.client_name = reader.read_string()
    dst.status = reader.read_uint8()
    dst.processing_percentage = reader.read_uint8()
    dst.result_code = reader.read_int32()
    return dst


def py_to_cdr_ServiceResponseHeader(src: ServiceResponseHeader) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_ServiceResponseHeader(writer, src)
    return writer.bytes()


def cdr_to_py_ServiceResponseHeader(cdr_payload) -> ServiceResponseHeader:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = ServiceResponseHeader()
    return cdr_body_to_py_ServiceResponseHeader(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_ServiceResponseHeader.js
import { ServiceResponseHeader } from './pdu_jstype_ServiceResponseHeader.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduServiceResponseHeaderConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { ServiceResponseHeader } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.request_id);
        writer.write_string(src.service_name);
        writer.write_string(src.client_name);
        writer.write_uint8(src.status);
        writer.write_uint8(src.processing_percentage);
        writer.write_int32(src.result_code);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { ServiceResponseHeader } dst
     * @returns { ServiceResponseHeader }
     */
    static cdr_body_to_js(reader, dst) {
        dst.request_id = reader.read_uint32();
        dst.service_name = reader.read_string();
        dst.client_name = reader.read_string();
        dst.status = reader.read_uint8();
        dst.processing_percentage = reader.read_uint8();
        dst.result_code = reader.read_int32();
        return dst;
    }

    /**
     * @param { ServiceResponseHeader } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { ServiceResponseHeader }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new ServiceResponseHeader());
    }
}
